great_expectations_cloud/agent/agent.py

```diff
@@ -130,6 +142,11 @@ def _listen(self) -> None:
             print("Received request to shutdown.")
         except (SubscriberError, ClientError):
             print("Connection to GX Cloud has encountered an error.")
+        except AuthenticationError:
+            # Retry with new credentials
+            self._config = self._get_config()
```

Comment:
> I thought that _listen calls it.

Maybe it should? The way this works right now, we will call `_get_config` immediately after the auth error which may not be what we want. Instead, we may want to wait to fetch/set new credentials. Maybe move `_get_config` out of `_init` and put it at the start of `_listen`?